src/Symfony/Component/HttpFoundation/InputBag.php

```diff
@@ -0,0 +1,130 @@
+        }
+
+        return $this === $value ? $default : $value;
+    }
+
+    /**
+     * Returns the inputs.
+     *
+     * @param string|null $key The name of the input to return or null to get them all
+     */
+    public function all(string $key = null): array
+    {
+        if (null === $key) {
+            return $this->parameters;
+        }
+
+        $value = $this->parameters[$key] ?? [];
+        if (!\is_array($value)) {
+            throw new Exception\BadRequestException(sprintf('Unexpected value for "%s" input, expecting "array", got "%s".', $key, \is_object($value) ? \get_class($value) : \gettype($value)));
+        }
+
+        return $value;
+    }
+
+    /**
+     * Replaces the current input values by a new set.
+     */
+    public function replace(array $inputs = [])
+    {
+        $this->parameters = [];
+        $this->add($inputs);
+    }
+
+    /**
+     * Adds input values.
+     */
+    public function add(array $inputs = [])
+    {
+        foreach ($inputs as $input => $value) {
+            $this->set($input, $value);
+        }
+    }
+
+    /**
+     * Sets an input by name.
+     *
+     * @param string|array $value
+     */
+    public function set(string $key, $value)
+    {
+        if (!is_scalar($value) && !\is_callable([$value, '__toString']) && !\is_array($value)) {
+            @trigger_error(sprintf('Passing "%s" as a 2nd Argument to "%s()" is deprecated since Symfony 5.1, pass a string or an array instead.', \is_object($value) ? \get_class($value) : \gettype($value), __METHOD__), E_USER_DEPRECATED);
+        }
+
+        $this->parameters[$key] = $value;
+    }
+
+    /**
+     * Filter key.
+     *
+     * @param mixed $default Default = null
+     * @param int   $filter  FILTER_* constant
+     * @param mixed $options Filter options
+     *
+     * @see https://php.net/filter-var
+     *
+     * @return mixed
+     */
+    public function filter(string $key, $default = null, int $filter = FILTER_DEFAULT, $options = [])
+    {
+        $value = $this->has($key) ? $this->all()[$key] : $default;
+
+        // Always turn $options into an array - this allows filter_var option shortcuts.
+        if (!\is_array($options) && $options) {
+            $options = ['flags' => $options];
+        }
+
+        // Add a convenience check for arrays.
+        if (\is_array($value)) {
+            if ((($flags = $options['flags'] ?? 0) & FILTER_REQUIRE_ARRAY) !== FILTER_REQUIRE_ARRAY || ($flags & FILTER_FORCE_ARRAY) !== FILTER_FORCE_ARRAY) {
+                @trigger_error(sprintf('Filtering an array value with "%s()" without passing the FILTER_REQUIRE_ARRAY flag is deprecated since Symfony 5.1, and will throw a "%s" exception in Symfony 6.0', __METHOD__, Exception\BadRequestException::class), E_USER_DEPRECATED);
+            }
+
+            if (!isset($options['flags'])) {
+                $options['flags'] = FILTER_REQUIRE_ARRAY;
```

Comment:
inside the previous if:
`$options['flags'] = ($options['flags'] ?? 0) | FILTER_REQUIRE_ARRAY;`

Reply:
This would be somehow a BC break. previously if flags is set, we don't modify it. the developer might not intend to support arrays, but a user has passed an array in the input

Reply:
OK - please still move it in the if before so that it's clear it should be removed on 6